crates/sui-test-validator/src/main.rs

```diff
@@ -1,263 +1,28 @@
-            )
-        }
-    };
-
-    if !result.transferred_gas_objects.is_empty() {
-        (StatusCode::CREATED, Json(result))
-    } else {
-        (StatusCode::INTERNAL_SERVER_ERROR, Json(result))
-    }
-}
-
-async fn faucet_batch_request(
-    Extension(state): Extension<Arc<AppState>>,
-    Json(payload): Json<FaucetRequest>,
-) -> impl IntoResponse {
-    let result = match payload {
-        FaucetRequest::FixedAmountRequest(FixedAmountRequest { recipient }) => {
-            state.faucet.batch_request_sui_coins(recipient).await
-        }
-        _ => {
-            return (
-                StatusCode::BAD_REQUEST,
-                Json(BatchFaucetResponse::from(FaucetError::Internal(
-                    "Input Error.".to_string(),
-                ))),
-            )
-        }
-    };
-    if result.task.is_some() {
-        (StatusCode::CREATED, Json(result))
-    } else {
-        (StatusCode::INTERNAL_SERVER_ERROR, Json(result))
-    }
-}
-
-async fn request_status(
-    Extension(state): Extension<Arc<AppState>>,
-    Path(id): Path<String>,
-) -> impl IntoResponse {
-    match Uuid::parse_str(&id) {
-        Ok(task_id) => {
-            let status = state.faucet.get_batch_send_status(task_id).await;
-            (StatusCode::CREATED, Json(status))
-        }
-        Err(e) => (
-            StatusCode::INTERNAL_SERVER_ERROR,
-            Json(BatchStatusFaucetResponse::from(FaucetError::Internal(
-                e.to_string(),
-            ))),
-        ),
-    }
+fn main() {
+    let info = "sui-test-validator binary has been deprecated in favor of sui start, which is a \
+                more powerful command that allows you to start the local network with more options. \
+There are three options that you can use to replace `sui-test-validator` binary:
+  * use `sui-pq` binary from a release archive from the GitHub releases page
+  * use the backward compatible `scripts/sui-test-validator.sh` script from the main Sui repository
+  * build from source with `indexer` feature enabled: `cargo build --bin sui --features indexer`
+
+We recommend to migrate to using sui start instead of using the script. To do so, use sui start \
+--help to see all the flags and options.
+
+To recreate the exact basic functionality of sui-test-validator (with no args passed), you must \
+use the following options
+  * --with-faucet --> to start the faucet server on the default host and port
+  * --force-regenesis --> to start the local network without persisting the state and from a new \
+genesis
+
+You can also use the following options to start the local network with more features:
+  * --with-indexer --> to start the indexer on the default host and port. Note that this requires \
+a Postgres database to be running locally, or you need to set the different options to connect to a \
+remote indexer database.
+  * --with-graphql --> to start the GraphQL server on the default host and port";
+
+    println!("{}", info);
```

Comment:
nit: The `info` variable is not doing much for you here, you could write this straight into the `println` format string.